sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -233,12 +235,28 @@ private async Task RunAsync(CancellationToken cancellationToken)
                 {
                     receivedEvents = await InnerConsumer.ReceiveAsync(Options.MaximumMessageCount, Options.MaximumReceiveWaitTime, cancellationToken).ConfigureAwait(false);
 
+                    using DiagnosticScope diagnosticScope = EventDataInstrumentation.ClientDiagnostics.CreateScope(DiagnosticProperty.EventProcessorProcessingActivityName);
+
+                    diagnosticScope.Start();
```

Comment:
can you start after you added links? 
Links are used to make sampling decision and it's done when span/activity start

Reply:
This is hard with how DiagnosticListener works (it uses Activity.Current https://github.com/open-telemetry/opentelemetry-dotnet/blob/cf33e8629083fac9f662f4fbe5ec3e4763808707/src/OpenTelemetry/Collector/DiagnosticSourceListener.cs#L66) so we need it started to be able to make AddLink call

Reply:
It's essential to make sampling decision consistent between links and processing. Otherwise under aggressive sampling tracing doesn't allow tracing things end-to-end.
Adding links on started activity would mean that the bridge has to start span on AddLink call rather than on start activity. This way start with payload (links) looks more promising

Reply:
Any ideas on how it implement this with what we have?

Reply:
I see a couple ways (that don't overcomplicate bridge)
1. pass links in the start event payload - i.e. bridge can use them before span starts. There is no contract so far for links, so good time to define one that suits user scenarios
2. pass links in the activity tags, before activity starts, this is crazy inefficient and we have to pass diagnostics-id around

I think 1 is the most reasonable.

Reply:
Good idea!